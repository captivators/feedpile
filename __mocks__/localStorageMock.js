const localStorageMock = (() => {
  let store = {
    "profile": '{"name":"faiz"}',
    id_token: ""
  };
  return {
    getItem(key) {
      return store[key]
    },
    setItem(key, value) {
      store[key] = value.toString()
    },
    clear() {
      store = {}
    }
  };
})();

global.localStorage = localStorageMock;
