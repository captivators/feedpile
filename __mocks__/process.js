const process = {
  "env" : {
    "AUTH_ID":"test",
    "AUTH_DOMAIN":"user_id"
  }
}
global.process = process;
