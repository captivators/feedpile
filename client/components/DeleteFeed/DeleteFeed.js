import React from 'react';
import Dialog from 'material-ui/Dialog';
import FlatButton from 'material-ui/FlatButton';
import RaisedButton from 'material-ui/RaisedButton';
import Checkbox from 'material-ui/Checkbox';
import {connect} from 'react-redux';
import userFeedsSelector from '../../selectors/user-feeds-selector';
import {toggleDeleteModal, deleteFeedsFromDb, setDisplayProgress} from '../../actions';

class DeleteFeed extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      feedsToDelete: []
    };
    this.handleCheck = this.handleCheck.bind(this)
  }

  handleCheck(isInputChecked, feedId) {
    if (isInputChecked) {
      this.state.feedsToDelete.push(feedId);
    } else if(!isInputChecked) {
      let remainingFeeds = this.state.feedsToDelete.filter(x => x !== feedId);
      this.state.feedsToDelete = remainingFeeds;
    }
  }

  render() {
    const actions = [
      <FlatButton
          label="Cancel"
          primary={true}
          onTouchTap={() => {
            this.props.toggleDeleteModal(false);
          }}
      />,
      <RaisedButton
          label="Submit"
          primary={true}
          onTouchTap={() => {
            this.props.setDisplayProgress(true);
            this.props.deleteFeedsFromDb(this.state.feedsToDelete,
                JSON.parse(localStorage.getItem('profile')).identities[0].user_id
            )
            this.props.toggleDeleteModal(false);
          }}
      />
    ];

    return (
        <div>
          {/* <RaisedButton label="Dialog With Date Picker" onTouchTap={this.handleOpen} /> */}
          <Dialog
              title="Choose the feeds you would like to delete."
              actions={actions}
              modal={false}
              open={this.props.openDeleteFeedModal}
              contentStyle={{width: '75%', maxWidth: '600px'}}
              onRequestClose={() => {
                this.props.toggleDeleteModal(false);
              }}
          >
            <div className="checkboxes">
              {this.props.userFeeds.map((feed, i) => {

                return (<Checkbox
                    key={i}
                    value={feed._id}
                    label={feed.name}
                    onCheck={(event, isInputChecked) => {
                      this.handleCheck(isInputChecked, feed._id);
                    }}
                />)
              })}
            </div>
          </Dialog>
        </div>
    );
  }

}
const mapStateToProps = (state) => {
  return {
    openDeleteFeedModal: state.openDeleteFeedModal,
    userFeeds: userFeedsSelector(state)
  }
};

export const Unwrapped = DeleteFeed;
export default connect(mapStateToProps, {toggleDeleteModal, deleteFeedsFromDb, setDisplayProgress})(DeleteFeed)
