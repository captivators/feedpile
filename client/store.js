import {createStore, compose, applyMiddleware} from 'redux';
import rootReducer from './root-reducer';
import logger from 'redux-logger';
import createSagaMiddleware from 'redux-saga';
import rootSaga from './root-saga';

const sagaMiddleware = createSagaMiddleware();

const store = createStore(rootReducer, compose (
  applyMiddleware(logger, sagaMiddleware)
));

sagaMiddleware.run(rootSaga);
export default store;
