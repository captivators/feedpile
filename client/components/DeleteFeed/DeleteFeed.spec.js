import '../../../__mocks__/localStorageMock'
import { Provider } from 'react-redux';
import React from 'react';
import { shallow, render } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import MuiThemeProvider from 'material-ui/styles/MuiThemeProvider';
import store from '../../store';
import { Unwrapped as UnwrappedDeleteFeed } from './DeleteFeed'
import getMuiTheme from '../../../node_modules/material-ui/styles/getMuiTheme';



test('DeleteFeed take a snapshot', () => {
  const muiTheme = getMuiTheme();
  const shallowWithContext = (node) => shallow(node, {context: {muiTheme}});
  const component = shallowWithContext(<UnwrappedDeleteFeed userFeeds={[]} store={ store }/>);
  const tree = shallowToJson(component);
  expect(tree).toMatchSnapshot();
});

// test('DeleteFeed renders modal dialog with checkboxes', () =>{
//   const feeds = [
//     {
//       _id: 'test id',
//       name: 'test name'
//     }
//   ]
//
//
//   const component = render(<MuiThemeProvider><UnwrappedDeleteFeed open={true} store={ store }
//                                                                    openDeleteFeedModal={false}
//                                                                   userFeeds={feeds}/></MuiThemeProvider>);
//   expect(component.find('Dialog').hasClass('.checkboxes')).toBe(true)
// })