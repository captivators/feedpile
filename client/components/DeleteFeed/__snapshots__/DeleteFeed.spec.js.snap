// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DeleteFeed take a snapshot 1`] = `
<div>
  <Dialog
    actions={
      Array [
        <FlatButton
          disabled={false}
          label="Cancel"
          labelPosition="after"
          labelStyle={Object {}}
          onKeyboardFocus={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          onTouchStart={[Function]}
          onTouchTap={[Function]}
          primary={true}
          secondary={false}
      />,
        <RaisedButton
          disabled={false}
          fullWidth={false}
          label="Submit"
          labelPosition="after"
          onTouchTap={[Function]}
          primary={true}
          secondary={false}
      />,
      ]
    }
    autoDetectWindowHeight={true}
    autoScrollBodyContent={false}
    contentStyle={
      Object {
        "maxWidth": "600px",
        "width": "75%",
      }
    }
    modal={false}
    onRequestClose={[Function]}
    repositionOnUpdate={true}
    title="Choose the feeds you would like to delete."
  >
    <div
      className="checkboxes"
    />
  </Dialog>
</div>
`;
